pkg/tls/tls.go

```diff
@@ -382,3 +384,39 @@ func toCASecretAndConfigmap(key *rsa.PrivateKey, cert *x509.Certificate, name st
 			},
 		}
 }
+
+// getServiceDNSName gets a cluster DNS name for a service given the services'
+// name and namespace.
+func getServiceDNSName(name, namespace string) string {
+	return fmt.Sprintf("%s.%s.svc.cluster.local", name, namespace)
+}
+
+// getPodDNSName gets a cluster DNS name for a pod given the pods' name and
+// namespace.
+func getPodDNSName(name, namespace string) string {
+	return fmt.Sprintf("%s.%s.pod.cluster.local", name, namespace)
+}
```

Comment:
I don't see where we use this function, but if we do (or plan to), it looks like the pod domain name convention uses the pod IP to build the name, so if we had a pod IP of `172.0.3.10` in namespace `default`, we could look it up with the following:
* `172-0-3-10.default.pod.cluster.local`
* `172-0-3-10.default.pod`

Should we change the `name` argument to `podIP` and do the conversion and expansion (see comment about service names) here?